cupy/cuda/device.pyx

```diff
@@ -73,12 +73,6 @@ def _get_attributes(device_id):
     d = {}
     for k, v in runtime_module.__dict__.items():
         if k.startswith('cudaDevAttr'):
-            if runtime._is_hip_environment:
-                # On ROCm 3.5.0 + gfx906, accessing these attributes leads to
-                # core dump (stack smashing detected)
-                if k in ('cudaDevAttrPciDeviceId',
-                         'cudaDevAttrTccDriver'):
-                    continue
```

Comment:
So it turns out this bug we hit is due to the enum mismatch between CUDA and HIP. After fixing it, the actual problematic attributes are `hipDeviceAttributeHdpMemFlushCntl` and `hipDeviceAttributeHdpRegFlushCntl`, see my comments in `cupy_backends/cuda/api/runtime.pxd`. The fix is shifted there (by not exposing them to Python).